protected/humhub/modules/admin/controllers/ApprovalController.php

```diff
@@ -121,4 +163,73 @@ public function actionDecline($id)
             'approveFormModel' => $model
         ]);
     }
+
+    /**
+     * @return string|\yii\console\Response|\yii\web\Response
+     * @throws HttpException
+     * @throws \Throwable
+     * @throws \yii\base\InvalidConfigException
+     */
+    public function actionBulkActions()
+    {
+        /** @var string $action */
+        $action = Yii::$app->request->post('action');
+        /** @var array $usersId */
+        $usersId = Yii::$app->request->post('ids');
+
+        if (!$action) {
+            $this->view->error(Yii::t('AdminModule.controllers_ApprovalController', 'Please select an action (approve or decline)'));
+            return $this->redirect(['index']);
+        }
+
+        if (!$usersId) {
+            $this->view->error(Yii::t('AdminModule.controllers_ApprovalController', 'Please select some users (tick the checkboxes)'));
+            return $this->redirect(['index']);
+        }
+
+        $model = new ApproveUserForm($usersId);
+
+        if ($action === self::ACTION_APPROVE) {
+            return $this->bulkApprove($model, $usersId);
+        }
+        if ($action === self::ACTION_DELINE) {
+            return $this->bulkDecline($model, $usersId);
+        }
+        throw new HttpException(400);
+    }
+
+    /**
+     * @param ApproveUserForm $model
+     * @return string|\yii\console\Response|\yii\web\Response
+     */
+    protected function bulkApprove(ApproveUserForm $model)
+    {
+        if($model->load(Yii::$app->request->post()) && $model->bulkApprove()) {
+            $this->view->success(Yii::t('AdminModule.controllers_ApprovalController', 'Users have been approved and emails have been sent'));
```

Comment:
- Category: `AdminModule.base`
- Text `The registrations were approved and the users were notified by email.`